src/main/java/org/joda/pa/MetaProperty.java

```diff
@@ -66,7 +67,7 @@
      * 
      * @return the type declaring the property, not null
      */
-    default Class<?> declaringType() {
+    default Class<B> declaringType() {
```

Comment:
Consider the hierarchy case here as well. The declaringType != bean type, but could be a superclass (maybe there is a case where it is completely unrelated?)